File: app/javascript/channels/room_channel.js
import consumer from "./consumer"

$(function(){
    const chatChannel = consumer.subscriptions.create({ channel: 'RoomChannel', group: $('#messages').data('room_id') }, {
      connected() {
    // Called when the subscription is ready for use on the server
  },

  received: function(data) {
    return $('#messages').append(data['message']);
  },

  speak: function(message,group_id) {
    return this.perform('speak', {message: message, group: group_id });
  }
});

  $(document).on('keypress', '[data-behavior~=room_speaker]', function(event) {
   if (event.key === 'Enter') {
    chatChannel.speak(event.target.value,$('#messages').data('room_id'));
    event.target.value = '';
    return event.preventDefault();
  }
  });
});

File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  def subscribed
    # room_channel.rbとroom_channel.jsでデータの送受信ができるようになる。
    stream_from "room_channel_#{params['group']}" 
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    # jsで実行されたspeakのmessageを受け取り、room_channelのreceivedにブロードキャストする
    Message.create! content: data['message'], customer_id: current_customer.id, group_id: params['group']
  end
end
